verification/curator-service/ui/src/components/BulkCaseForm.tsx

```diff
@@ -0,0 +1,144 @@
+    > {
+    constructor(props: BulkCaseFormProps) {
+        super(props);
+        this.state = {
+            statusMessage: '',
+        };
+    }
+    // Array<any> is the type used by the source library for this data.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async uploadData(data: Array<any>, fileInfo: IFileInfo): Promise<void> {
+        for (const c of data) {
+            try {
+                await axios.put('/api/cases', {
+                    caseReference: {
+                        sourceId: c['sourceId'],
+                        sourceEntryId: c['sourceEntryId'],
+                    },
+                    demographics: {
+                        sex: c['sex'],
+                        ageRange: {
+                            start: c['ageRangeStart'],
+                            end: c['ageRangeEnd'],
+                        },
+                    },
+                    location: {
+                        country: c['country'],
+                        administrativeAreaLevel1: c['administrativeAreaLevel1'],
+                        administrativeAreaLevel2: c['administrativeAreaLevel2'],
+                        administrativeAreaLevel3: c['administrativeAreaLevel3'],
+                        query: c['locationQuery'],
+                        geometry: {
+                            latitude: c['latitude'],
+                            longitude: c['longitude'],
+                        },
+                        geoResolution: c['geoResolution'],
+                        name: c['locationName'],
+                    },
+                    events: [
+                        {
+                            name: 'confirmed',
+                            dateRange: {
+                                start: c['dateConfirmedStart'],
+                                end: c['dateConfirmedEnd'],
+                            },
+                        },
+                    ],
+                    sources: [
+                        {
+                            url: c['url'],
+                        },
+                    ],
+                    revisionMetadata: {
+                        revisionNumber: 0,
+                        creationMetadata: {
+                            curator: this.props.user.email,
+                            date: new Date().toISOString(),
+                        },
+                    },
+                });
+                this.setState({ statusMessage: 'Success!' });
+            } catch (e) {
+                if (e.response) {
+                    this.setState({ statusMessage: e.response.data });
+                } else if (e.request) {
+                    this.setState({ statusMessage: e.request });
+                } else {
+                    this.setState({ statusMessage: e.message });
+                }
+            }
+        }
+    }
+
+    render(): JSX.Element {
+        const { classes } = this.props;
+        const papaparseOptions = {
```

Comment:
hmm can't tell if this is a typo or intentional :P what is the 'papa' prefix for? It sounds like a sports announcer introducing parseOptions 😁

Reply:
Ha, yeah -- the react-csv-reader package uses a popular file parsing package called Papa Parse. Seems pretty canonical to name the options for react-csv-reader this way.